cockpit/src/main/java/com/polidea/cockpit/paramsedition/layout/CockpitLayout.kt

```diff
@@ -0,0 +1,192 @@
+package com.polidea.cockpit.paramsedition.layout
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ValueAnimator
+import android.content.Context
+import android.support.annotation.IdRes
+import android.support.v4.widget.ViewDragHelper
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+import android.view.animation.AccelerateDecelerateInterpolator
+import android.widget.FrameLayout
+import com.polidea.cockpit.R
+
+
+internal class CockpitLayout @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0)
+    : FrameLayout(context, attrs, defStyleAttr) {
+
+    var startDrag: Boolean = false
+    private var state = State.EXPANDED
+    private lateinit var draggableView: View
+    private lateinit var collapsedDraggableViewProperties: DraggableViewProperties
+
+    private val dragHelper: ViewDragHelper = ViewDragHelper.create(this, 1f, DragHelperCallback())
+
+    private val topPositionExpanded = 0
+    private var currentTopPosition = topPositionExpanded
+    private val minDialogHeight = resources.getDimension(R.dimen.cockpit_draggable_min_height).toInt()
+    private val cockpitMargin = resources.getDimension(R.dimen.cockpit_draggable_margin).toInt()
+
+    fun setDraggableView(@IdRes draggableViewId: Int) {
+        draggableView = findViewById(draggableViewId)
+    }
+
+    fun expand() {
+        state = State.EXPANDING
+        animateDraggableViewToNextState(collapsedDraggableViewProperties.margin, 0,
+                collapsedDraggableViewProperties.top.toFloat(), 0f,
+                collapsedDraggableViewProperties.height, height, State.EXPANDED)
+    }
+
+    fun collapse() {
+        state = State.COLLAPSING
+        animateDraggableViewToNextState(0, collapsedDraggableViewProperties.margin,
+                0f, collapsedDraggableViewProperties.top.toFloat(),
+                height, collapsedDraggableViewProperties.height, State.COLLAPSED)
+    }
+
+    fun resize(height: Int) {
+        val params = draggableView.layoutParams
+        params.height = Math.max(height, minDialogHeight)
```

Comment:
You should also check if resized view doesn't overlap the screen. Otherwise we can expand it to much. The limit of expansion should be the bottom margin.